### ISSUE TITLE
Optimize autoupdater to download only necessary parts

### ISSUE THREAD

**Issue by Eggypapa** [wabbajack-tools/wabbajack#583]
**Is your feature request related to a problem? Please describe.**
Every time there is an update for Wabbajack, the application will automatically download and install the update. When it does so, it seems to be downloading everything all over again. 
At least for me, this takes 5-10 minutes because of my slow internet. And because of my slow internet, I often need to relaunch Wabbajack over the course of several days to finish off a download. Each time having to download almost the whole application again.

**Describe the solution you'd like**
Ideally, the solution would be for Wabbajack to only download the specific files it needs to for the update. Perhaps leaving all the redists in a shared folder or something like that?
Based off looking at the update folders in winmerge, only a handful of files actually change from version to version.

**Describe alternatives you've considered**
Alternative solutions could be a defer update button or investigating something like file-by-file patching (bsdiff for example).

**Additional context**
Making this change would be really positive in my opinion, as it would reduce the amount users need to download and probably reduce the amount of data you need to serve.

Thanks for your time and hard work.

**Comment by Noggog** [comment]
Hey Eggy!

Just some clarification questions:
When you say that wabbajack has an update, do you mean wabbajack the app itself, or one of the modlists?

Assuming you mean wabbajack itself is what is updating that causes the issue, when you say ` it seems to be downloading everything all over again` you then mean it has to download your preferred modlist's mods all over again, and doesn't seem to be reusing old downloads?

**Comment by Eggypapa** [comment]
Hello Noggog!

Sorry for not being particularly clear, when I say wabbajack has an update I mean the wabbajack application itself, not specific modlists.

For example, on the 25th of Feb I downloaded wabbajack version 0.9.21.0 from github and went to run it, however before it could run it needed to update to 0.9.22.0.
A few days later on the 27th of Feb, I launched Wabbajack again to continue downloading my modlist, however before I could start downloading the modlist I needed to wait for wabbaajck to update to version 0.9.22.1

Each time one of these updates download, it downloads a lot of unnecessary files that are already on the users computers.
For example, the update from 0.9.22.0 only contained 5mb of changed files from version 0.9.21.0, however it still appears to have downloaded the full 150mb-ish update and has unpacked to half a gig on my HDD.

Between the three versions of wabbajack I have downloaded at the moment they in total only have around maybe 7-8 different files from version to version. However, they are taking up more than a gigabyte of space on my HDD of mostly identical files.

This is obviously working as intended at the moment and isn't a bug, as being able to quickly get users to roll back to an old version makes a lot of sense. However, wabbajack could still achive that goal by simply splitting all the resource DLL files of into a shared version agnostic folder as they contribute to most of the update size and don't change from version to version.
Something like this maybe?
![image](https://user-images.githubusercontent.com/56150254/75497665-19b20200-5a19-11ea-88a0-a67e0af7f766.png)

Sorry for the giant braindump there, I just feel reducing the size of update downloads would benifit everybody and got onto a roll.
